pallets/phala/src/stakepool.rs

```diff
@@ -642,8 +650,13 @@ pub mod pallet {
 			ensure_signed(origin)?;
 			Self::ensure_pool(pid)?;
 			let sub_account: T::AccountId = pool_sub_account(pid, &worker);
-			let (orig_stake, slashed) = mining::Pallet::<T>::reclaim(sub_account)?;
+			let (orig_stake, slashed) = mining::Pallet::<T>::reclaim(sub_account.clone())?;
 			Self::handle_reclaim(pid, orig_stake, slashed);
+			// A successful relcaim will settle all the stake. We don't care about the preimage of
+			// the sub-account anymore. So we feel safe to delete the preimage. Also, this method
+			// is the only entrance to trigger the underlying reclaim. So we can release the
+			// storage now.
+			SubAccountPreimages::<T>::remove(sub_account);
```

Comment:
Do you think we should have a event here? so indexer easy to know that the preimage deleted

Reply:
Better to track the other relevant events (mining start & reclaimed)